File: go/quick_sort/ith_min_test.go
package quick_sort

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestIthMin_RandomCases(t *testing.T) {
	testIthMin_RandomCases(t, 1000, 100)
}

func TestIthMin_RandomCases_LargeArrays(t *testing.T) {
	testIthMin_RandomCases(t, 100, 100_000)
}

func testIthMin_RandomCases(t *testing.T, testCount, maxN int) {
	for range testCount {
		td := newRandomTestDataN(maxN)

		expected := td.expectedArray[td.randomIndex]
		actual := ithMin(td.array, td.randomIndex)

		if actual != expected {
			assert.FailNowf(t, "",
				"ithMin(%v, %v):\n  Actual: %v\nExpected: %v",
				td.array, td.randomIndex, actual, expected,
			)
		}
	}
}

File: python/quick_sort/ith_min_test.py
from ith_min import ith_min
from quick_sort_test_data import RandomTestData


def test_ith_min_random_cases():
	ith_min_random_cases(1000, 100)


def test_ith_min_random_cases_large_arrays():
	ith_min_random_cases(100, 10_000)


def ith_min_random_cases(test_count: int, max_n: int):
	for _ in range(test_count):
		td = RandomTestData(max_n)

		want = td.want[td.random_index]
		got = ith_min(td.array, td.random_index)

		assert got == want, \
			f'ith_min({td.array}, {td.random_index}):\nwant {want}\ngot {got}'
